### ISSUE TITLE
[Bug]: ilike is broken

### ISSUE THREAD

**Issue by fengttt** [matrixorigin/matrixone#7711]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

mysql> select * from log_info where message like '%panic%';
Empty set (12.53 sec)

mysql> select * from log_info where message ilike '%panic%';
ERROR 1064 (HY000): SQL parser error: You have an error in your SQL syntax; check the manual that corresponds to your MatrixOne server version for the right syntax to use. syntax error at line 1 column 42 near " ilike '%!p(MISSING)anic%!'(MISSING)";
mysql>

### Expected Behavior

ilike should work whenever like works.

**Comment by aressu1985** [comment]
test tomorrow

**Comment by aressu1985** [comment]
fixed:46a417daf4b55b93a35682995033ed18b8f1d096
Database changed
mysql> show tables;
+------------------+
| Tables_in_system |
+------------------+
| statement_info   |
| rawlog           |
| log_info         |
| error_info       |
| span_info        |
+------------------+
5 rows in set (0.01 sec)

mysql>  select * from log_info where message like '%panic%';
Empty set (0.17 sec)

mysql>  select * from log_info where message ilike '%panic%';
Empty set (0.19 sec)

mysql>
